### ISSUE TITLE
Starting `Cachex` via `Supervisor.start_link` 

### ISSUE THREAD

**Issue by lucavenir** [whitfin/cachex#383]
Hello there,

I *really* hope I'm not wasting anyone's time here, but I simply went and read the "welcome page/guide" of this (awesome!) package and I've got myself a papercut.

[I read](https://hexdocs.pm/cachex/overview.html#setting-up) from the docs, in regards of starting `Cachex` in a supervision tree:
```ex
children = [
  {Cachex, [:my_cache]},     # with default options
  {Cachex, [:my_cache, []]}  # with custom options
]
```

So I try this in my phoenix application; I copy-paste the first line in my list of `children`:
```ex
children = [
  ...,
  {Cachex, [:my_cache]},
]
```

But, `Supervisor.start_link` gives this error:
```
** (Mix) Could not start application my_app: MyApp.Application.start(:normal, []) returned an error: shutdown: failed to start child: Cachex
    ** (EXIT) :invalid_name
```

Whereas, if I write my cache name like below, no error is raised on startup:
```ex
{Cachex, name: :my_cache},
```

Which makes sense, AFAIK supervisor wants a keyword list as a second argument of the tuple.

But... am I doing this right? Am I missing something?

**Comment by lucavenir** [comment]
By the way I've read #380 before opening this issue. I've updated my dependency to `4.0.1` but the behavior is identical.

**Comment by whitfin** [comment]
Hi @lucavenir! 

At a glance, yes, the docs for this are wrong and need to be updated. I can double check when I get a moment, just in case!

**Comment by whitfin** [comment]
@lucavenir instead of changing the docs, I decided to just make sure that the form shown in the README also works - I did that in #384. I also added checks to make sure this is tested going forward!

It should be released in v4.0.2 if you want to double check that it's all good for you now!
